File: BankApplicationAPI/BankApplicationAPI/Models/TransactionLog.cs
namespace BankApplicationAPI.Models;

public partial class TransactionLog
{
    public int? TransactionId { get; set; }

    public DateTime? TransactionDate { get; set; }

    public byte? TransactionTypeId { get; set; }

    public decimal? TransactionAmount { get; set; }

    public decimal? NewBalance { get; set; }

    public int? AccountId { get; set; }

    public string? EmployeeId { get; set; }

    public string? CustomerId { get; set; }

    public virtual Account? Account { get; set; }

    public virtual Customer? Customer { get; set; }

    public virtual Employee? Employee { get; set; }

    public virtual TransactionType? TransactionType { get; set; }
}

File: BankApp/src/app/Modules/TransactionLog.ts
import { Account } from "./Account";
import { Customer } from "./Customer";
import { Employee } from "./Employee";
import { TransactionType } from "./TransactionType";

export interface TransactionLog {
    transactionId?: number | null;
    transactionDate?: Date | null;
    transactionTypeId?: number | null;
    transactionAmount?: number | null;
    newBalance?: number | null;
    accountId?: number | null;
    employeeId?: string | null;
    customerId?: string | null;
    account?: Account | null;
    customer?: Customer | null;
    employee?: Employee | null;
    transactionType?: TransactionType | null;
}